### PR TITLE
users_file_management

### PR DESCRIPTION
**Pull request by interExportVuk**
Authenitcation and test module

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -1,22 +1,118 @@
...
 @app.route('/insert', methods=['POST'])
+@requires_auth
 def insert_client():
-    status = busy_wait_listener(Job(1, 'Insert user', action_insert_user(User('User1', '', True, True))))
-    return status
+    if session['write_permission'] is True:
+        status = busy_wait_listener(Job(1, 'Insert user', action_insert_user('User1', '', True, True)))
+        return status
+    else:
+        return 'INSUFFICIENT  PERMISSION'
```
Comment (iblagojevic) [inline app.py]: You can define all error responses in sort of `constants` file and pull values from there.
Also, have you considered having corresponding http response header in these cases (e.g. 401, etc.)

File: app.py [hunk 2]
```diff
@@ -1,22 +1,118 @@
...
+    secret_key = private_key_generator(10)
+    c = Client(secret_key, True, True)
+    db_session.add(c)
+    try:
+        db_session.flush()
+        db_session.refresh(c)
+        db_session.commit()
+        return jsonify(id=c.id, key=secret_key)
+    except:
+        db_session.rollback()
```
Comment (iblagojevic) [inline app.py]: Is there a need to maybe add `finally` block with `db_session.close()` here?

File: app.py (hunk 1 above)
Comment (interExportVuk) [inline app.py]: That is a great idea! I shall add constants file as well as response file. That way clients can better understand server responses.

File: app.py (hunk 2 above)
Comment (interExportVuk) [inline app.py]: I think that it is not necessary because it will make @app.teardown_appcontext function usless.

**Pull request closed without merging**